centreon/src/Core/Notification/Application/UseCase/AddNotification/AddNotification.php

```diff
@@ -0,0 +1,424 @@
+    }
+
+    /**
+     * @param AddNotificationRequest $request
+     *
+     * @throws \Throwable
+     *
+     * @return NotificationResource[]
+     */
+    private function createResourceObjects(AddNotificationRequest $request): array
+    {
+        if ($request->resources === []) {
+            throw NotificationException::emptyArrayNotAllowed('resource');
+        }
+
+        $newResources = [];
+        /** @var array{
+         *      type:string,
+         *      events:int,
+         *      ids:int[],
+         *      includeServiceEvents:int
+         * } $resourceData
+         */
+        foreach ($request->resources as $resourceData) {
+            $resourceIds = array_unique($resourceData['ids']);
+
+            if (count($resourceIds) === 0) {
+                continue;
+            }
+
+            $repository = $this->resourceRepositoryProvider->getRepository($resourceData['type']);
+
+            if ($this->user->isAdmin()) {
+                // Assert IDs validity without ACLs
+                $existingResources = $repository->exist($resourceIds);
+            } else {
+                // Assert IDs validity with ACLs
+                $accessGroups = $this->readAccessGroupRepository->findByContact($this->user);
+                $existingResources = $repository->existByAccessGroups($resourceIds, $accessGroups);
+            }
+
+            $difference = new BasicDifference($resourceIds, $existingResources);
+            $missingResources = $difference->getRemoved();
+            if ([] !== $missingResources) {
+                $this->error(
+                    'Invalid ID(s) provided',
+                    ['propertieName' => 'resources', 'propertieValues' => array_values($missingResources)]
+                );
+
+                throw NotificationException::invalidId('resource.ids');
+            }
+
+            // If multiple resources with same type are defined, only the last one of each type is kept
+            $newResources[$repository->resourceType()] = new NotificationResource(
+                $repository->resourceType(),
+                $repository->eventEnum(),
+                array_map((fn($resourceId) => new NotificationGenericObject($resourceId, '')), $resourceIds),
+                ($repository->eventEnumConverter())::fromBitmask($resourceData['events']),
+                $resourceData['includeServiceEvents']
+                    ? NotificationServiceEventConverter::fromBitmask($resourceData['includeServiceEvents'])
+                    : []
+            );
+        }
+
+        $totalResources = 0;
+        foreach ($newResources as $newResource) {
+            $totalResources += $newResource->getResourcesCount();
+        }
+        if ($totalResources <= 0) {
+            throw NotificationException::emptyArrayNotAllowed('resource.ids');
+        }
+
+        return $newResources;
+    }
+
+    /**
+     * @param int $notificationId
+     * @param NotificationResource[] $newResources
+     */
```

Comment:
Missing `@throws` in several locations in this class like here.
```suggestion
    /**
     * @param int $notificationId
     * @param NotificationResource[] $newResources
     *
     * @throws \Throwable
     */
```